subsys/pm/device_runtime.c

```diff
@@ -104,13 +105,13 @@ int pm_device_runtime_get(const struct device *dev)
 
 	SYS_PORT_TRACING_FUNC_ENTER(pm, device_runtime_get, dev);
 
-	if (!k_is_pre_kernel()) {
-		(void)k_mutex_lock(&pm->lock, K_FOREVER);
+	if (!atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_ENABLED)) {
+		ret = -ENOTSUP;
+		goto end;
 	}
 
-	if (!pm->enable) {
-		ret = -ENOTSUP;
-		goto unlock;
+	if (!k_is_pre_kernel()) {
+		(void)k_mutex_lock(&pm->lock, K_FOREVER);
```

Comment:
This change also breaks thread safety:

1. Thread A calls `pm_device_runtime_get()`,
2. Atomic test succeeds (device is enabled)
3. Thread A is preempted, Thread B becomes active.
4. Thread B calls `pm_device_runtime_disable()` and succeeds
5. Thread B is preempted, Thread A becomes active
6. Thread A continues on a device with disabled runtime PM (WRONG)